src/CoSELayout.js

```diff
@@ -1439,12 +1560,23 @@ CoSELayout.prototype.tileNodes = function (nodes, minWidth) {
     centerY: 0
   };
 
+  if (tilingSortBy) {
+    organization.idealRowWidth = this.calcIdealRowWidth(nodes, favorHorizontalDim);
+  }
+
+  var getCompVal = function getCompVal(n) {
+    if (organization.idealRowWidth) {
+      return tilingSortBy(n.id);
+    }
+    return n.rect.width * n.rect.height;
+  };
+
   // Sort the nodes in ascending order of their areas
   nodes.sort(function (n1, n2) {
```

Comment:
Here, it would be better to keep sorting in descending order. Especially sorting in descending order when the node area is used (default) generally gives better results. The comment from the past might be written wrong. We may also add an explanation for `tilingSortBy` option in extension side to clarify that it sorts in descending order based on the returned value.

Reply:
Let's apply the first one to be consistent between two options. If user wants to sort in ascending order, he/she can do that easily by multiplying the value with -1 before returning it.

Reply:
@hasanbalci I guess with this option there would be a lot of cases where the return value of the function will not be a number. If it is a string I think there may be a need for reversing the string instead of multiplying by -1. I actually do not know if it would be an issue for some users as the strings get longer and number of nodes increase. 

If it sounds okay I would add an extra option like ``tilingSortByOrder`` which is ``descending`` by default. Node areas will be sorted in descending order anyways this option is only to work together with ``tilingSortBy`` option.

Reply:
@hasanbalci maybe before moving on I can make another suggestion about this part. Maybe we can consider having another option like ``tilingCompareBy`` instead of these two that I mentioned (``tilingSortBy`` and  ``tilingSortByOrder``). 

It may work similar to optional ``compareFunction`` of javascript's ``Array.prototype.sort()`` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).

I think that advantage of this option over the other two options we discussed maybe enabling users to have more customizable sorting. For example when users are working with mix of strings and numbers javascript's grater than operator will return true for ``"a2" > "a11"``. However, I suspect some users may prefer "a11" considered bigger than "a2" for sorting. Also, maybe having just one option would be simpler to manage.

What do you think?